internal/dag/builder.go

```diff
@@ -518,6 +521,18 @@ func (b *Builder) computeHTTPProxy(proxy *projcontour.HTTPProxy) {
 			svhost.Secret = sec
 			svhost.MinProtoVersion = annotation.MinProtoVersion(proxy.Spec.VirtualHost.TLS.MinimumProtocolVersion)
 
+			if proxy.Spec.VirtualHost.TLS.EnableFallbackCertificate && !isBlank(b.FallbackCertificate) {
+				secretName := splitSecret(b.FallbackCertificate, "projectcontour")
+				sec, err = b.lookupSecret(secretName, validSecret)
+				if err != nil {
+					sw.SetInvalid("FallbackCertificate requested, but Certificate Secret %q is invalid or missing", b.FallbackCertificate)
```

Comment:
Here, and in the case above, we should put the resolved secret name (i.e. the result of `splitSecret`) in the status message, since that tells the user exactly which object had the problem.

Reply:
The reason I left it out is that it leaks the namespace/name of the secret used.  Since it's a wildcard, I didn't know if we wanted to keep that from users.